ENIGMAsystem/SHELL/Graphics_Systems/General/GSfont.cpp

```diff
@@ -89,39 +89,39 @@ unsigned draw_get_valign(){
 #ifdef DEBUG_MODE
   #include "Widget_Systems/widgets_mandatory.h"
   #define get_font(fnt,id,r) \
-    if (id < -1 or (size_t(id+1) > enigma::font_idmax) or !fontstructarray[id]) { \
+    if (!enigma::fonts.exists(id)) { \
```

Comment:
Why arent all these macros in ass array?

Reply:
Actually, these could all just be thrown out. AssetArray as it works now only uses macros internally.

However, there's been a dispute over how AssetArray chooses to crash in run mode, and return a sentinel if you ask for oob in debug mode. Josh and Rusky have both expressed that macros are better because they can just cancel the function and return rather than return a stack sentinel. That would mean creating a new family of functions in the AssetArray header and those would replace these get font macros.

I haven't decided whether to do that or not and thus didn't rush into throwing out these old macros.

Reply:
If the functionality is covered, let's. If needed, we can replace these with more general macros that use a method within `AssetArray` to perform debug checking and report errors. But honestly, `AssetArray` should be equipped to report the errors, as-is.